src/resolver/nodeModuleLookup.ts

```diff
@@ -117,104 +125,93 @@ export function findTargetFolder(props: IResolverProps, parsed: IModuleParsed):
-  // extract entry point
-
-  // extract target if required
-  if (parsed.target) {
-    const parsedLookup = fileLookup({ fileDir: folder, target: parsed.target });
-    if (!parsedLookup) {
-      return { error: `Failed to resolve ${props.target} in ${parsed.name}` };
-    }
-
-    result.targetAbsPath = parsedLookup.absPath;
 
-    if (isBrowser && json.browser && typeof json.browser === 'object') {
-      const override = handleBrowserField(pkg, parsedLookup.absPath);
-      if (override) {
-        result.targetAbsPath = override;
-        parsedLookup.customIndex = true;
-      }
-    }
-
-    result.isEntry = false;
-    result.targetFuseBoxPath = makeFuseBoxPath(folder, result.targetAbsPath);
-  } else {
-    const entryFile = getFolderEntryPointFromPackageJSON({ isBrowserBuild: isBrowser, json: json });
-
-    const entryLookup = fileLookup({ fileDir: folder, target: entryFile });
+  const isEntry = !target;
+  const targetResolver = (isUser && props.tsTargetResolver) || fileLookup;
+  const resolved = targetResolver({ fileDir: folder, target: target || "", isBrowserBuild: isBrowser });
+  if (!resolved || !resolved.fileExists) {
+    const spec = target ? `"${target}"` : "an entry point";
+    return { error: `Failed to resolve ${spec} in package "${moduleName}"` };
+  }
+  let targetAbsPath = resolved.absPath;
 
-    if (!entryLookup.fileExists) {
-      return {
-        error: `Failed to resolve an entry point in package ${parsed.name}. File ${entryFile} cannot be resolved.`,
-      };
+  const checkBrowserOverride = isBrowser && json.browser && typeof json.browser === 'object'
+  if (checkBrowserOverride) {
+    const override = handleBrowserField(pkg, resolved.absPath);
+    if (override) {
+      targetAbsPath = override;
     }
+  }
 
-    pkg.entryAbsPath = entryLookup.absPath;
-    pkg.entryFuseBoxPath = makeFuseBoxPath(folder, entryLookup.absPath);
-
-    result.isEntry = true;
-
-    result.targetAbsPath = pkg.entryAbsPath;
-    result.targetFuseBoxPath = pkg.entryFuseBoxPath;
-
-    if (isBrowser && json.browser && typeof json.browser === 'object') {
-      const override = handleBrowserField(pkg, entryLookup.absPath);
-      if (override) {
-        //result.targetFuseBoxPath =
-        result.targetAbsPath = override;
-        pkg.entryAbsPath = override;
-        result.targetFuseBoxPath = makeFuseBoxPath(folder, override);
-        pkg.entryFuseBoxPath = result.targetFuseBoxPath;
+  const targetFuseBoxPath = makeFuseBoxPath(folder, targetAbsPath);
 
-        entryLookup.customIndex = true;
-      }
-    }
+  if (isEntry) {
+    pkg.entryAbsPath = targetAbsPath;
+    pkg.entryFuseBoxPath = targetFuseBoxPath;
   }
-  result.targetExtension = path.extname(result.targetAbsPath);
 
-  return result;
+  const targetExtension = path.extname(targetAbsPath);
+
+  return {
+    isEntry,
+    targetAbsPath,
+    targetFuseBoxPath,
+    targetExtension,
+    meta: pkg,
+    isUser,
```

Comment:
shall we give it a better name? I took a while for me to figure it out

Reply:
Maybe `isUserModule`?

The point is that this module is one of the user's own modules.  I.e. one that the user will edit.  And so it should be treated differently during caching when scanning for changes.

But I'm certainly open to suggestions on this one too.  But I'm going to submit this one later in a separate PR anyway